### PR TITLE
test(x/ecocredit): marketplace integration tests

### PR DESCRIPTION
**Pull request by ryanchristo**
## Description

Closes: #627

This pull request adds integration tests for the marketplace functionality.

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added `!` to the type prefix if API or client breaking change
- [ ] targeted the correct branch (see [PR Targeting](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [ ] followed the guidelines for [building modules](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules)
- [ ] included the necessary unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] added a changelog entry to `CHANGELOG.md`
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed `!` in the type prefix if API or client breaking change
- [ ] confirmed all author checklist items have been addressed 
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

File: x/ecocredit/server/msg_server.go [hunk 1]
```diff
@@ -747,7 +750,10 @@ func (s serverImpl) Buy(goCtx context.Context, req *ecocredit.MsgBuy) (*ecocredi
 			if creditsRemaining.IsZero() {
 
 				// delete sell order if no remaining credits
-				s.sellOrderTable.Delete(ctx, sellOrder.OrderId)
+				err = s.sellOrderTable.Delete(ctx, sellOrder.OrderId)
+				if err != nil {
+					return nil, err
+				}
```
Comment (aleem1314) [inline x/ecocredit/server/msg_server.go:756]:
```suggestion
				if err := s.sellOrderTable.Delete(ctx, sellOrder.OrderId); err != nil {
					return nil, err
				}
```

**Review by aleem1314 — APPROVED** [review]
lgtm

File: x/ecocredit/server/testsuite/tx.go [hunk 2]
```diff
@@ -0,0 +1,587 @@
...
+			_, err := s.msgClient.UpdateSellOrders(s.ctx, &ecocredit.MsgUpdateSellOrders{
+				Owner:   tc.owner,
+				Updates: tc.updates,
+			})
+
+			if tc.wantErr {
+				require.Error(err)
+				require.Contains(err.Error(), tc.expErr)
+			} else {
+				require.NoError(err)
```
Comment (technicallyty) [inline x/ecocredit/server/testsuite/tx.go:290]: should we check that the updates applied to the sell order here?

Comment (ryanchristo) [reply x/ecocredit/server/testsuite/tx.go:303]: added in https://github.com/regen-network/regen-ledger/pull/626/commits/dc6d6273a6a4dfb67f85ec6581e1b434aee1b500

File: x/ecocredit/server/testsuite/tx.go [hunk 3]
```diff
@@ -0,0 +1,587 @@
...
+			}
+		})
+	}
+}
+
+func (s *IntegrationTestSuite) TestScenarioCreateBuyOrders() {
```
[the remark this answers is not in the document]
Comment (ryanchristo) [reply x/ecocredit/server/testsuite/tx.go:324]: added in https://github.com/regen-network/regen-ledger/pull/626/commits/dc6d6273a6a4dfb67f85ec6581e1b434aee1b500

File: x/ecocredit/server/testsuite/tx.go [hunk 4]
```diff
@@ -0,0 +1,672 @@
...
+			if tc.wantErr {
+				require.Error(err)
+				require.Contains(err.Error(), tc.expErr)
+				require.Equal(coinBalanceBefore, coinBalanceAfter)
+				require.Equal(creditBalanceBefore, creditBalanceAfter)
+			} else {
+				require.NoError(err)
+				require.NotNil(res.BuyOrderIds)
+				require.NotEqual(coinBalanceBefore, coinBalanceAfter)
+				require.NotEqual(creditBalanceBefore, creditBalanceAfter)
```
Comment (technicallyty) [inline x/ecocredit/server/testsuite/tx.go:554]: would prefer to have exacts checked here to ensure tx math is correct

**Review by technicallyty — APPROVED** [review]
LGTM - one last nit 👍🏻

File: x/ecocredit/server/testsuite/tx.go (hunk 4 above)
Comment (ryanchristo) [reply x/ecocredit/server/testsuite/tx.go]: added in https://github.com/regen-network/regen-ledger/pull/626/commits/e7fe6a9e4099aeeceacb344b0ffc401b2192b03f

**Pull request merged**